README.md

```diff
@@ -2,39 +2,118 @@
+
+The project focuses on ensuring that your application realises the full potential of the [Reactive Manifesto](http://www.reactivemanifesto.org/), 
+while delivering a high productivity development environment, and seamless production deployment experience.
+
+## Features
+
+ScaleCube is designed as an embeddable library for the Java VM. It is built in a modular way where each module independently 
+implements a useful set of functionality and provides foundation for higher level features by abstracting away lower level concerns.
+
+Next is described modules in the top to bottom order from the higher level features to the low level components.
+
+### MICROSERVICES
+
+ScaleCube Services provides a low latency Reactive Microservices library for peer-to-peer service registry and discovery 
+based on gossip protocol ad without single point-of-failure or bottlenecks.
+
+User Guide:
+
+* [Services Overview] (http://scalecube.io/services.html)
+* [Defining Services] (http://scalecube.io/user-reference/services/DefineService.html)
+* [Implementing services] (http://scalecube.io/user-reference/services/ServiceImplementation.html)
+* [Provisioning Clustered Services] (http://scalecube.io/user-reference/services/ProvisionClusterServices.html)
+* [Consuming services] (http://scalecube.io/user-reference/services/ConsumingServices.html)
+
+### CLUSTER
+
+ScaleCube Cluster is a lightweight decentralized cluster membership, failure detection, and gossip protocol library. 
+It provides an implementation of [SWIM](http://www.cs.cornell.edu/~asdas/research/dsn02-swim.pdf) cluster membership protocol for Java VM.
+It is an efficient and scalable weakly-consistent distributed group membership protocol based on gossip-style communication between the 
+nodes in the cluster. Read my [blog post](http://www.antonkharenko.com/2015/09/swim-distributed-group-membership.html) with distilled 
+notes on the SWIM paper for more details.
+
+It is using a [random-probing failure detection algorithm](http://www.antonkharenko.com/2015/08/scalable-and-efficient-distributed.html) which provides 
+a uniform expected network load at all members. 
+The worst-case network load is linear O(n) for overall network produced by running algorithm on all nodes and constant network 
+load at one particular member independent from the size of the cluster.
+
+ScaleCube Cluster implements all improvements described at original SWIM algorithm paper, such as gossip-style dissemination, suspicion mechanism 
+and time-bounded strong completeness of failure detector algorithm. In addition to that we have introduced support of additional SYNC mechanism 
+in order to improve recovery of the cluster from network partitioning events.
+  
+Using ScaleCube Cluster as simple as few lines of code:
+ 
+``` java
+// Start cluster node Alice as a seed node of the cluster, listen and print all incoming messages
+ICluster alice = Cluster.joinAwait();
```

Comment:
i suggest not to have this here but have some nice user guide how to use the cluster
and links here to user guide and also on the website